usdt-impl/src/no-linker.rs

```diff
@@ -138,9 +138,9 @@ fn extract_probe_records_from_section() -> Result<Section, crate::Error> {
     let data = unsafe {
         let start = (&dtrace_probes_start as *const usize) as usize;
         let stop = (&dtrace_probes_stop as *const usize) as usize;
-        std::slice::from_raw_parts(start as *const u8, stop - start)
+        std::slice::from_raw_parts_mut(start as *mut u8, stop - start)
     };
-    process_section(data)
+    process_section(data, /* register = */ true)
```

Comment:
FWIW rust-analyzer drops in the `register` annotation

Reply:
You mean that RA adds that comment for you? That's helpful! I don't use RA, so thought it'd be useful for humans as well. Is it more harmful than helpful?